fmskill/comparison.py

```diff
@@ -1149,22 +1150,65 @@ def __copy__(self):
     def copy(self):
         return self.__copy__()
 
-    def _model2obs_interp(self, obs, mod_df):
+    def _model2obs_interp(self, obs, mod_df, max_gap):
         """interpolate model to measurement time"""
-        df = self._interp_df(mod_df, obs.time)
+        df = self._interp_df(mod_df, obs.time, max_gap)
         # mod_ds.interp_time(obs.time).to_dataframe()
         df[self.obs_name] = obs.values
         return df
 
     @staticmethod
-    def _interp_df(df, new_time):
+    def _parse_max_gap(modeldatalist, max_model_gap):
+        if not isinstance(max_model_gap, list):
+            max_model_gap = [max_model_gap]
+        if len(max_model_gap) != len(modeldatalist):
+            if len(max_model_gap) == 1:
+                max_model_gap *= len(modeldatalist)
+            else:
+                raise ValueError("Length of max_model_gap must match length of modeldata.")
+        return max_model_gap
+
+    @staticmethod
+    def _interp_df(df, new_time, max_gap):
         assert df.index.is_unique
         assert new_time.is_unique
-        new_df = (
+        limit = None
+        interpolation=True
+        #Long list of checks for the minimum value that can be interpolated if different values are given, since,
+        #for a fair model comparison, all model results must share same index
+        if max_gap ==[None]:
+            max_gap=None
+        # if multi model then max_gap is list of values or None's
+        if type(max_gap)==list:
+            #We assume a large number to find the minimum `max_gap` 
+            min_gap=1e19
+            for t in max_gap:
+                if t is not None:
+                    min_gap=min(min_gap,t)
+            max_gap=min_gap
+            if max_gap==1e19:
+                max_gap=None 
+        # Calculate the `mode` of dt
+        if (len(new_time) > 0) and interpolation and (max_gap is not None):
+            ix = new_time[1:] - new_time[0:-1]
+            vals, counts = np.unique(ix, return_counts=True)
+            mode_value_ix = np.argwhere(counts == np.max(counts))
+            dt_meas = vals[mode_value_ix][0][0] / np.timedelta64(1, "s")
+            limit = int(max_gap / dt_meas)
```

Comment:
Is this pandas solution actually robust enough? A fix limit? `dt_meas` will often be non-equidistant, right?